torchx/test/support/torchx_case.ex

```diff
@@ -0,0 +1,20 @@
+defmodule Torchx.Case do
+  @moduledoc """
+  Test case for tensor assertions
+  """
+
+  use ExUnit.CaseTemplate
+
+  using do
+    quote do
+      import Torchx.Case
+    end
+  end
+
+  defmacro assert_tensor({:==, _, [left, right]}) do
+    quote do
+      assert Nx.backend_transfer(unquote(left), Nx.BinaryBackend) ==
+               Nx.backend_transfer(unquote(right), Nx.BinaryBackend)
```

Comment:
Wouldn't an `allclose?` or `isclose?` fit better here? See: https://pytorch.org/docs/stable/testing.html

Reply:
Yeah, but we don't have it implemented yet! Should I withhold this PR and implement `all?` + `allclose?` instead?

Reply:
@seanmor5 I've refactored this to use Nx.all_close as suggested :)